am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java

```diff
@@ -59,11 +70,40 @@ void whenRowExistsAndDoesNotHaveReadPermissionsShouldReturnEnvelopeWithoutData()
     }
 
     @Test
-    void whenNoRowExistsShouldReturnNull() {
+    void whenThereAreNoAccessRecordsShouldReturnNull() {
         String nonExistingUserId = "ijk";
         String nonExistingResourceId = "lmn";
 
-        FilterResourceResponse result = ams.filterResource(nonExistingUserId, nonExistingResourceId, DATA);
+        FilterResourceResponse result = ams.filterResource(
+            nonExistingUserId, ROLE_NAMES, createResource(nonExistingResourceId));
+
+        assertThat(result).isEqualTo(FilterResourceResponse.builder()
+            .resourceId(resourceId)
+            .data(null)
+            .permissions(ImmutableMap.of(JsonPointer.valueOf(""), CREATE_PERMISSION))
```

Comment:
How does it work? There is not pre-configuration logic. Roles setup from other tests? Do we truncate roles table before tests?